src/test/java/com/springboot/userservices/service/UserServiceTest.java

```diff
@@ -0,0 +1,170 @@
+  {
+    fixture.givenUserIsInitialized(1, "Eric", "Cartman", "CEO");
+    fixture.whenUserIsAdded();  //to reduce redundancy in Fixture class
+    fixture.whenUserIsDeleted();
+    fixture.thenCountOfUserIs(0);
+  }
+
+  @Test(expected = IndexOutOfBoundsException.class)
+  public void testRemoveUser_userDoesNotExist_doesNotRemoveUser_throwsException()
+  {
+    fixture.givenUserIsInitialized(1, "Eric", "Cartman", "CEO"); // User count=1
+    fixture.whenUserIsAdded();
+    fixture.givenUserIsReInitializedAfterAddition(10, "Scully", "Cartman", "CEO"); // assigning same object to a new reference
+    fixture.whenUserIsDeleted();  // will look for Eric in usersArray, won't find it so won't delete, count remains 1
+    fixture.thenCountOfUserIs(1);
+    fixture.thenUserServiceThrewIndexOutOfBoundException();
+  }
+
+  @Test
+  public void testUpdateUser_updatesUserSuccessfully()
+  {
+    fixture.givenUserIsInitialized(1, "Eric", "Cartman", "CEO");
+    fixture.whenUserIsAdded();
+    fixture.whenUserIsUpdated("Scully", "Cartman", "Manager");
+    fixture.thenCheckIfUserIsUpdatedSuccessfully();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testUpdateUser_enteredInvalidAttributes_doesNotUpdateUser_throwsException()
+  {
+    fixture.givenUserIsInitialized(1, "Eric", "Cartman", "CEO");
+    fixture.whenUserIsAdded();
+    fixture.whenUserIsUpdated("", "", "");
+    fixture.thenCheckIfUserIsUpdatedSuccessfully();
+    fixture.thenUserServiceThrewIllegalArgumentException();
+  }
+
+  private class Fixture
+  {
+    private UserModel mockedUser = Mockito.mock(UserModel.class);
+    private UserService userService = new UserService(mockedUser);
+    private Exception exception = new Exception();
+
+    public void givenUserIsInitialized(int id, String firstName, String lastName, String employeeType)
+    {
+      this.mockedUser.setId(id);
+      this.mockedUser.setFirstName(firstName);
+      this.mockedUser.setLastName(lastName);
+      this.mockedUser.setEmployeeType(employeeType);
+      if (id < 0)
+      {
+        when(mockedUser.getId()).thenReturn(-1);   //specifying Mocked object to put value in getId as -1
+      }
+    }
+
+    public void whenUserIsAdded()
+    {
+      try
+      {
+        this.userService.addUser(mockedUser);
+      }
+      catch (IllegalArgumentException e)
+      {
+        this.exception = e;
+      }
+    }
+
+    public void thenCountOfUserIs(int expectedCount)
+    {
+      assertEquals(expectedCount, userService.getUsers().size());
+    }
+
+    public void thenUserServiceThrewIllegalArgumentException()
+    {
+      assertEquals(IllegalArgumentException.class, exception.getClass());    //check if exception thrown is same as given
+      throw new IllegalArgumentException();
+    }
+
+    public void thenUserServiceThrewIndexOutOfBoundException()
+    {
+      assertEquals(IndexOutOfBoundsException.class, exception.getClass());
+      throw new IndexOutOfBoundsException();
```

Comment:
if you are already asserting exception, then we dont need to throw it again and test it in @Test, one assertion is enough. 
The reason is, if we change the behaviour of exception, we need to modify it at two places. Also this is making us write more code.